pyvista/core/pointset.py

```diff
@@ -1862,59 +1858,105 @@ class StructuredGrid(_vtk.vtkStructuredGrid, PointGrid, StructuredGridFilters):
 
     Can be initialized in one of the following several ways:
 
-    - Create empty grid
-    - Initialize from a vtk.vtkStructuredGrid object
-    - Initialize directly from the point arrays
+    * Create empty grid.
+    * Initialize from a filename.
+    * Initialize from a ``vtk.vtkStructuredGrid`` object.
+    * Initialize directly from one or more :class:`numpy.ndarray`. See the
+      example or the documentation of ``uinput``.
 
-    See _from_arrays in the documentation for more details on initializing
-    from point arrays
+    Parameters
+    ----------
+    uinput : str, pathlib.Path, vtk.vtkStructuredGrid, numpy.ndarray, optional
+        Filename, dataset, or array to initialize the structured grid from. If
+        a filename is passed, pyvista will attempt to load it as a
+        :class:`StructuredGrid`. If passed a ``vtk.vtkStructuredGrid``, it will
+        be wrapped as a deep copy.
+
+        If a :class:`numpy.ndarray` is provided and ``y`` and ``z`` are empty,
+        this array will define the points of this StructuredGrid. Set the
+        dimensions with :attr:`StructuredGrid.dimensions`.
+
+        Otherwise, this parameter be loaded as the ``x`` points and ``y`` and
+        ``z`` points must be set. The shape of this array defines the shape of
+        the structured data and the shape should be ``(dimx, dimy,
+        dimz)``. Missing dimensions are assumed to be ``1``.
```

Comment:
```suggestion
        Otherwise, this parameter will be loaded as the ``x`` points, and ``y``
        and ``z`` points must be set. The shape of this array defines the shape
        of the structured data and the shape should be ``(dimx, dimy,
        dimz)``. Missing trailing dimensions are assumed to be ``1``.
```